### ISSUE TITLE
Ability to disable NAT for IPv6 GUA

### ISSUE THREAD

**Issue by nem64** [containers/netavark#560]
Good day,
So I'm in a pretty weird situation right now. I created a bridge network in Podman with a GUA /56 subnet hoping to give each container a publicly available address. However, I have no way of instructing Podman to disable NAT for this specific bridge network. My current stop gap workaround is to just prepend a `RETURN` rule for all the subnet in the NAT chain as so `ip6tables -t nat -I POSTROUTING -s XXXX:YYYY:ZZZZ:000::/56 -j RETURN`

I want to do this in a clean way without having to wrangle iptables. How feasible is this?

**Comment by mheon** [comment]
I don't think this is possible right now, but should not be difficult to add - just need to detect if the address is global scoped and decline to add NAT rules if so.

**Comment by nem64** [comment]
The ip6tables rule is working wonderfully right now so it's not a priority but it'd be appreciated if it would could get integrated into netavark and eventually podman

**Comment by Luap99** [comment]
Will port forwarding still work without the NAT? I understand that it is not really needed with global addresses but if not we might cause regressions to user who need it.

**Comment by mheon** [comment]
I think it ought to - though we might need to create some of the normal NAT chains to make sure they exist (just not the ones that actually match and translate).

**Comment by nem64** [comment]
It's also possible to make NAT opt-out like how Docker does it with `com.docker.network.bridge.enable_ip_masquerade`. This way you'll have to explicitly enable an option to disable NAT for a specific network. Should theoretically prevent regressions instead of doing it automatically by detecting ULA/GUA

**Comment by RobertMe** [comment]
Having an option to disable masquerading, like Docker has, would be nice anyway. I'm currently investigating whether I could use a different output interface with Podman, but this doesn't seem possible (except when using slirp4netns). While in Docker this is possible by disabling masquerading and manually applying an SNAT rule (as for example described [here](https://medium.com/@havloujian.joachim/advanced-docker-networking-outgoing-ip-921fc3090b09) ). I guess this would also work in Podman when overruling the masquarade using an earlier (/higher priority) SNAT rule. But obviously is a hack as well same as written in this tickets start post.

**Comment by nbarrientos** [comment]
It'd be indeed very beneficial if traffic with source a global IPv6 address was (by default, perhaps) forwarded without any kind if SNAT. I think it's becoming more and more common that people get routable IPv6 prefixes from upstream providers which are directly distributed to containers. Having to fiddle with auto-generated firewall rules as @nem64 is suggesting is a working solution but it's indeed far from ideal.

**Comment by Luap99** [comment]
Contributions welcome. I think the first step would to add an option to disable masquerading https://github.com/containers/netavark/issues/831, I guess one for both ipv4/6 and then a specific one for each ipv4/6.

We can certainly discus having no masquerading as default for global ipv6 addresses but I guess due the risk of that being a breaking chnage we would have to wait until we make a major version bump.

**Comment by ranile** [comment]
[This blog post about docker](https://blog.apnic.net/2025/05/19/how-to-configure-routed-ipv6-in-docker/) is a good reference for prior art on this topic
